Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift

```diff
@@ -643,7 +644,7 @@ extension NIOThrowingAsyncSequenceProducer {
                     failure: failure
                 )
 
-            case .finished:
+            case .finished(iteratorInitialized: false):
                 // It is strange that an iterator is created after we are finished
                 // but it can definitely happen, e.g.
                 // Sequence.init -> source.finish -> sequence.makeAsyncIterator
```

Comment:
I think we should change the state here, shouldn't we?

Reply:
Yep good catch. So annoying that you can't write tests for `fatalError`s easily.